runtime/convertValues_test.go

```diff
@@ -1012,8 +1013,716 @@ func importAndExportValuesFromScript(t *testing.T, script string, arg cadence.Va
+		},
+		{
+			label:         "UInt32",
+			typeSignature: "UInt32",
+			exportedValue: cadence.NewUInt32(42),
+		},
+		{
+			label:         "UInt64",
+			typeSignature: "UInt64",
+			exportedValue: cadence.NewUInt64(42),
+		},
+		{
+			label:         "UInt128",
+			typeSignature: "UInt128",
+			exportedValue: cadence.NewUInt128(42),
+		},
+		{
+			label:         "UInt256",
+			typeSignature: "UInt256",
+			exportedValue: cadence.NewUInt256(42),
+		},
+		{
+			label:         "Word8",
+			typeSignature: "Word8",
+			exportedValue: cadence.NewWord8(42),
+		},
+		{
+			label:         "Word16",
+			typeSignature: "Word16",
+			exportedValue: cadence.NewWord16(42),
+		},
+		{
+			label:         "Word32",
+			typeSignature: "Word32",
+			exportedValue: cadence.NewWord32(42),
+		},
+		{
+			label:         "Word64",
+			typeSignature: "Word64",
+			exportedValue: cadence.NewWord64(42),
+		},
+		{
+			label:         "Fix64",
+			typeSignature: "Fix64",
+			exportedValue: cadence.Fix64(-123000000),
+		},
+		{
+			label:         "UFix64",
+			typeSignature: "UFix64",
+			exportedValue: cadence.UFix64(123000000),
+		},
+		{
+			label:         "StoragePath",
+			typeSignature: "StoragePath",
+			exportedValue: cadence.Path{
+				Domain:     "storage",
+				Identifier: "foo",
+			},
+			skipExport: true,
+		},
+		{
+			label:         "PrivatePath",
+			typeSignature: "PrivatePath",
+			exportedValue: cadence.Path{
+				Domain:     "private",
+				Identifier: "foo",
+			},
+			skipExport: true,
+		},
+		{
+			label:         "PublicPath",
+			typeSignature: "PublicPath",
+			exportedValue: cadence.Path{
+				Domain:     "public",
+				Identifier: "foo",
+			},
+			skipExport: true,
+		},
+		{
+			label:         "Address",
+			typeSignature: "Address",
+			exportedValue: cadence.NewAddress([8]byte{0, 0, 0, 0, 0, 1, 0, 2}),
+		},
+
+		// TODO: Enable below once https://github.com/onflow/cadence/issues/712 is fixed.
+		// TODO: Add a malformed argument test for capabilities
+		//{
+		//	label:         "Capability",
+		//	typeSignature: "Capability<&Foo>",
+		//	exportedValue: cadence.Capability{
+		//		Path: cadence.Path{
+		//			Domain:     "public",
+		//			Identifier: "bar",
+		//		},
+		//		Address:    cadence.NewAddress([8]byte{0, 0, 0, 0, 0, 1, 0, 2}),
+		//		BorrowType: "Foo",
+		//	},
+		//},
+
+		// TODO: enable once https://github.com/onflow/cadence/issues/491 is fixed.
+		//{
+		//	label:         "Type",
+		//	typeSignature: "Type",
+		//	exportedValue: cadence.TypeValue{
+		//		StaticType: "Foo",
+		//	},
+		//},
+
+	}
+
+	testArgumentPassing := func(test argumentPassingTest) {
+
+		t.Run(test.label, func(t *testing.T) {
+
+			t.Parallel()
+
+			returnSignature := ""
+			returnStmt := ""
+
+			if !test.skipExport {
+				returnSignature = fmt.Sprintf(": %[1]s", test.typeSignature)
+				returnStmt = "return arg"
+			}
+
+			script := fmt.Sprintf(
+				`pub fun main(arg: %[1]s)%[2]s {
+
+					if !arg.isInstance(Type<%[1]s>()) {
```

Comment:
Good idea to check both argument validation (value conforms to static type) and also check dynamic type conformance 👍